### DISCUSSION TITLE
Why doesn't Airflow have a cancelled state for DAG runs/TaskInstance?

### DISCUSSION THREAD

**Discussion by TakawaAkirayo** [apache/airflow#43461] [General]
Currently I can archive this by updating the state as 'Failed'.

https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/update_dag_run_state 

However, this makes it difficult for me to distinguish between an actual failure and a cancellation.
Would introducing a 'cancelled' state impact the existing design, and would it be worthwhile to do so?

**Comment by potiuk** [comment]
> Would introducing a 'cancelled' state impact the existing design, and would it be worthwhile to do so? 

It's been considered in the past I think, but if you want to propose it and describe the changes and it's impact - feel free. Likely this requires an improvement proposal, and some analysis what it means to internals of Airflow. Also that might be a good thing to consider for Airflow 3 - because we stopped adding features to Airflow 2 and are full-on working on Airflow 3. So if you want to propose somethign like this - it's the right thing to do. Look here https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+3.0 - where information on our bi-weekly dev calls is explained, you can start discussion on devlist.

As Airflow is developed by volunteers, you either need to do all the work on analysis on what it means yourself and propose it, or engage some people enough and enthuse them to work on it to make it happen.

**Reply by TakawaAkirayo** [reply]
@potiuk Thank you for the guidance! Let me take a look and study the history and current situation first. I’m excited to join the community discussion.

**Reply by alispa** [reply]
I am wondering the same. In my personal use case I would like to cancel/abort/kill a task or even the whole DAG run. Currently I have to find workarounds to do it on my own. There is an 8 year old [stackoverflow question about the same topic](https://stackoverflow.com/questions/43631693/how-to-stop-kill-airflow-tasks-from-the-ui). I was really surprised that even 8 years later there was no such feature.
